### ISSUE TITLE
STDERR not outputted by default

### ISSUE THREAD

**Issue by itay-grudev** [fish-shell/fish-shell#2115]
fish does not forward STDERR to STDOUT. Of course if I append ```^^&1``` to the command that I am running everything is ok, but that doesn't seem to work by default.

**Example:**
```bash
user@Computer ~> rm non_existing_file
user@Computer ~ [1]> rm non_existing_file ^&1
rm: non_existing_file: No such file or directory
user@Computer ~ [1]>
```

Platform:
Mac OS X 10.9 (Mavericks)

Terminal:
Terminal.app (OS X Default)

**Comment by ridiculousfish** [comment]
Might be the same as #2084. Unfortunately we'd need reproducible steps to make progress on this. Let me know if you figure out what's triggering it.

**Comment by itay-grudev** [comment]
I solved the issue and I documented how in #2084.
